src/ecalc/libraries/libecalc/common/libecalc/input/yaml_types/temporal_model.py

```diff
@@ -1,5 +1,11 @@
-from datetime import datetime
-from typing import MutableMapping, TypeVar, Union
+from typing import Dict, TypeVar, Union
+
+from pydantic import ConstrainedStr
+
+
+class DatetimeString(ConstrainedStr):
+    regex = "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])$"
+
 
 TModel = TypeVar("TModel")
-TemporalModel = Union[TModel, MutableMapping[datetime, TModel]]
+TemporalModel = Union[TModel, Dict[DatetimeString, TModel]]
```

Comment:
We need to figure out how to do this, the schema for datetime does not enforce anything on the key, while a constr gives us the same regex as before. It still does not give a validation error in the editor, while in test it does, I'm assuming this is because we have 'additionalProperties' in the pydantic generated temporal model while the old schemas does not.

Reply:
ah, ok, so we do not really want the "additionalProperties" then?

Reply:
correct, at least if monaco-editor can handle that schema